pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -1054,37 +1054,20 @@ private static int parseSql(char[] p_sql, int i, StringBuilder newsql, boolean s
             break;
           } else if (c == '{') { // start of an escape code?
             if (i + 1 < len) {
-              char next = p_sql[i + 1];
-              char nextnext = (i + 2 < len) ? p_sql[i + 2] : '\0';
-              if (next == 'd' || next == 'D') {
-                state = SqlParseState.ESC_TIMEDATE;
-                i++;
-                newsql.append("DATE ");
-                break;
-              } else if (next == 't' || next == 'T') {
-                state = SqlParseState.ESC_TIMEDATE;
-                if (nextnext == 's' || nextnext == 'S') {
-                  // timestamp constant
-                  i += 2;
-                  newsql.append("TIMESTAMP ");
-                } else {
-                  // time constant
-                  i++;
-                  newsql.append("TIME ");
+              SqlParseState[] availableStates = SqlParseState.values();
```

Comment:
> afaik it's worth caching.

Well, the difference seems to be just 5 bytes: 1632-1627.
Feel free to cache or non-cache